plans/provision.pp

```diff
@@ -1,28 +1,46 @@
 # @summary Provision new PE cluster to The Cloud
 #
 plan pecdm::provision(
-  TargetSpec                                $targets              = get_targets('peadm_nodes'),
-  Enum['xlarge', 'large', 'standard']       $architecture         = 'standard',
-  Enum['development', 'production', 'user'] $cluster_profile      = 'development',
-  String[1]                                 $version              = '2019.8.5',
-  String[1]                                 $console_password     = 'puppetlabs',
-  Integer                                   $compiler_count       = 1,
-  Optional[String[1]]                       $ssh_pub_key_file     = undef,
-  Optional[Integer]                         $node_count           = undef,
-  Optional[String[1]]                       $instance_image       = undef,
-  Optional[String[1]]                       $stack                = undef,
-  Array                                     $firewall_allow       = [],
-  Hash                                      $extra_peadm_params   = {},
-  Hash                                      $extra_terraform_vars = {},
-  Boolean                                   $replica              = false,
-  Boolean                                   $stage                = false,
+  TargetSpec                                    $targets              = get_targets('peadm_nodes'),
+  Enum['xlarge', 'large', 'standard']           $architecture         = 'standard',
+  Enum['development', 'production', 'user']     $cluster_profile      = 'development',
+  Enum['direct', 'bolthost']                    $download_mode        = 'direct',
+  String[1]                                     $version              = '2019.8.10',
+  String[1]                                     $console_password     = 'puppetlabs',
+  Integer                                       $compiler_count       = 1,
+  Optional[String[1]]                           $ssh_pub_key_file     = undef,
+  Optional[Integer]                             $node_count           = undef,
+  Optional[String[1]]                           $instance_image       = undef,
+  Optional[String[1]]                           $stack                = undef,
+  Optional[Variant[String[1],Array[String[1]]]] $subnet               = undef,
+  Optional[String[1]]                           $subnet_project       = undef,
```

Comment:
So... we already ask for `$project` in our inputs. Is there ever a scenario where there will be a difference between the project a subnet is part of, and the value a user provides for `$project`?

Generalizing the question back to the Terraform module, is there any reason to call this input parameter "subnet_project" there, rather than just "project"?

Reply:
Opinion: I don't think we should expose this as a parameter here. I think we should ask the user only for a project, and assume/require that subnets specified be part of that project. But, I'm making some assumptions about GCP in taking that stance.

@ody if there were a common or reasonable use case where the two could be different, I could be swayed from that.

Reply:
One use case for this is if you had a VPN network shared with a project. You would have a single project that contains all of the managed networks that a limited number of admins can access and then those networks get shared out to other projects for consumption. This is the configuration that we use in some of our GCP projects and need to specify a subnetwork_project, network, and subnetwork. However, we currently do that through the `extra_terraform_vars`. I am not sure if sharing networks is common enough to have specific parameters for it.

An example of how these can be specified with the current parameters is below.
```
    "extra_terraform_vars": {
      "network"            : "https://www.googleapis.com/compute/v1/projects/networkingproject/global/networks/shared",
      "subnetwork"         : "sharednetwork",
      "subnetwork_project" : "networkingproject",
        "labels"             : {
          "owner": "jarret",
      }
    }
```

Reply:
The `subnet_project` parameter is optional already. If you use an existing subnet which is shared from another project then you set it, else if the existing subnet is in your project than you do not. If 80% of deployments to existing subnets are in project then as is, 80% of deployments will not have to set the parameter.

Reply:
@jarretlavallee Suggestion of passing this piece of data via the `extra_terraform_vars` I am also OK with. I do think attaching to a shared network is in the minority for most executions of the `pecdm::provision` plan...but...since I am on a call with PayPal doing cloud deployment stuff and they have a fairly established hybrid GCP environment, I asked. It is their opinion that in the context of deploying PE, it'll near always be that they'll be attaching to a subnet shared from another project.

Reply:
Sounds like:
* It's already optional. If the existing subnet is in the same project , you can specify `subnet` and leave `subnet_project` undefined. Yes? Then great!
* From at least one major customer, we're hearing that in the majority of use case for specifying an existing subnet, the subnet will come from a different project. So, that's an argument for leaving it exposed as a `pecdm` parameter.

Reply:
The `extra_terraform_vars` workflow is possible, but more difficult since the user would need to know the terraform variables and formatting required to send the parameters. 
It seems like there are arguments for both sides, with one having a real customer attached to it who will always use a subnet project configuration. That seems like a good justification for adding it as a parameter, especially since it will help with the usage of this code past the initial deployment and new users coming online.